### PR TITLE
Clarify intro and ffmpeg versions

### PR DESCRIPTION
**Pull request by pushfoo**
### Changes

* Make it clear that FFmpeg is now optional in the intro
* Make intro generally clearer and more concise
* Add notes on FFmpeg and licensing

### CODE REVIEW

**Review by caffeinepills — CHANGES_REQUESTED** [review]
I submitted some areas I thought could use addressing. Other than that, looks good overall. Thanks

File: doc/programming_guide/media.rst [hunk 1]
```diff
@@ -3,24 +3,27 @@
...
-If FFmpeg is not present, pyglet will at a minimum be able to play WAV files
-only. Depending on the OS, an additional limited amount of compressed formats
-may also be supported, but only WAV is guaranteed (see "Supported media types
-" below). the This may be sufficient for many applications that require only a
-small number of short sounds, in which case those applications need not distribute FFmpeg.
+pyglet can load and play many audio and video formats, often with
+support for surround sound and video effects.
 
-.. _FFmpeg: https://www.ffmpeg.org/download.html
+In practice, WAV files always load and MP3 usually does. The exact
```
[the remark this answers is not in the document]
Comment (pushfoo) [reply doc/programming_guide/media.rst]: Reverted to be closer to the original.

File: doc/programming_guide/media.rst [hunk 2]
```diff
@@ -336,13 +339,28 @@ To install PyOgg, please see their `installation guide on readthedocs.io
...
-FFmpeg requires an external dependency, please see installation instructions
-in the next section below.
+.. _FFmpeg's license overview: https://www.ffmpeg.org/legal.html
+
+.. note:: The most recent pyglet release can use FFmpeg 4.X, 5.X, or 6.X
+
+          See :ref:`guide-media-ffmpeginstall` to learn more.
+
+FFmpeg is best when you need to support the maximum number of formats
+and encodings. Its main downsides are:
```
[the remark this answers is not in the document]
Comment (pushfoo) [reply doc/programming_guide/media.rst]: I've changed this to be closer to the original.

File: doc/programming_guide/media.rst [hunk 3]
```diff
@@ -336,13 +339,28 @@ To install PyOgg, please see their `installation guide on readthedocs.io
...
+See :ref:`guide-ffmpeg-licenses` to learn more.
+
+Supported Formats
+"""""""""""""""""
 
-With FFmpeg, many common and less-common formats are supported. Due to the
-large number of combinations of audio and video codecs, options, and container
-formats, it is difficult to provide a complete yet useful list. Some of the
-supported audio formats are:
+It is impossible to list all its features here due to the large number
```
[the remark this answers is not in the document]
Comment (pushfoo) [reply doc/programming_guide/media.rst]: Updated to be closer to the original. :+1:

File: doc/programming_guide/media.rst [hunk 4]
```diff
@@ -336,13 +339,28 @@ To install PyOgg, please see their `installation guide on readthedocs.io
...
 ^^^^^^
-FFmpeg requires an external dependency, please see installation instructions
-in the next section below.
+.. _FFmpeg's license overview: https://www.ffmpeg.org/legal.html
+
+.. note:: The most recent pyglet release can use FFmpeg 4.X, 5.X, or 6.X
+
+          See :ref:`guide-media-ffmpeginstall` to learn more.
+
+FFmpeg is best when you need to support the maximum number of formats
```
[the remark this answers is not in the document]
Comment (caffeinepills) [reply doc/programming_guide/media.rst:350]: No GStreamer for video.

**Review by caffeinepills — APPROVED** [review]

**Pull request merged**